onnxruntime/contrib_ops/cpu/bert/attention.cc

```diff
@@ -339,55 +509,84 @@ Status Attention<T>::Compute(OpKernelContext* context) const {
         const int qkv_index = static_cast<int>(i % 3);
 
         int input_offset = batch_index * sequence_length * input_hidden_size;
-        int weights_offset = qkv_index * hidden_size + head_index * head_size;
+
         T* qkv_dest = QKV[qkv_index];
-        int qkv_offset = (batch_index * num_heads_ + head_index) * (sequence_length * head_size);
+        int qkv_offset = 0;
+        int head_size_passed_in = 0;
+
+        if (qkv_index == 0) {
```

Comment:
nit: It might be faster to define a constant before the for loop:
   const int QKV_HEAD_SIZE[3] = {q_head_size, k_head_size, v_head_size};
Then use int head_size = QKV_HEAD_SIZE[qkv_index] here.